### PR TITLE
Fix git commit message verification script

### PR DESCRIPTION
**Pull request by joelsmith**
Problems with old version:
* only looked at the top commit
* always assumed merges to master branch
* used upper-case shell variables
* didn't handle PR numbers for upstream
* required extraneous "openshift: " in commit messages
* provided no feedback on what messages should look like

### CODE REVIEW

**Comment by joelsmith** [comment]
I added some test commits so that this PR could test itself with some bad input. For these three commits:

```
UPSTREAM: <carry>: Fix git commit message verification script
test, ignore
UPSTREAM: <carry>: add whitespace to script output
```

I get the following output from the script:
```
examining commits between 95d5f0927b9347635f1720d6748503415e6921cb and 07831cf5065559527b88d32dc4cc6b38579a9156

UPSTREAM: <carry>: add whitespace to script output
Git history in this PR doesn't conform to set commit message standards. Offending commit message is:
test, ignore

commit messages should look like one of:
UPSTREAM: <carry>: message  (commits that should be carried indefinitely)
UPSTREAM: <drop>: message   (commits that should be dropped on the next upstream rebase)
UPSTREAM: 1234: message     (commits that should be carried until an upstream rebase includes upstream PR 1234)
```

File: hack/verify_history.sh [hunk 1]
```diff
@@ -14,22 +14,34 @@
...
-  exit 1
-fi
+read -d '' help_message << EOF
 
-git log master.. --pretty=%s --no-merges | while read -r; do
-  if ! [[ "$REPLY" =~ ^$PREFIX ]]; then
+commit messages should look like one of:
+UPSTREAM: <carry>: message  (commits that should be carried indefinitely)
+UPSTREAM: <drop>: message   (commits that should be dropped on the next upstream rebase)
+UPSTREAM: 1234: message     (commits that should be carried until an upstream rebase includes upstream PR 1234)
```
Comment (JoelSpeed) [inline hack/verify_history.sh]: ❤️ ❤️ ❤️ This is brilliant. I recently worked on this project for the first time and was so confused when I saw the old error message, this is 10000 times better

**Comment by frobware** [comment]
/lgtm

Nice improvements! Thanks.

**Pull request merged**